processing/src/main/java/org/apache/druid/query/QueryContext.java

```diff
@@ -54,7 +54,7 @@
 {
   private static final QueryContext EMPTY = new QueryContext(null);
 
-  private final Map<String, Object> context;
+  private Map<String, Object> context;
```

Comment:
A large amount of code depends on the fact that this class is immutable. There was a big project a couple months back to clean up problems that arose when an earlier version was mutable. 

If we wish to clear the context, we should do so by replacing one instance of this class with the `EMPTY` context.

Reply:
This instance replace is not easily possible as the current `Query` interface does not support replace, only override (where the keys are preserved). 
To solve this currently, I am using the ObjectMapper to create an ObjectNode from the query object and manually clear the "context" key, which is working fine. I also tried creating a Mockito's spy object and overriding the context methods, however, that was failing the equals and the serde tests (presumably because the child classes won't be the same)